tap_integration_tests/tests/showcase.rs

```diff
@@ -0,0 +1,722 @@
+        let future_2: Pin<Box<dyn Future<Output = Result<(), jsonrpsee::core::Error>> + Send>> =
+            client_2.request("request", (id_2, receipt_2));
+        let (result_1, result_2) = join!(future_1, future_2);
+
+        match result_1 {
+            Ok(()) => {}
+            Err(e) => panic!("Error making receipt request: {:?}", e),
+        }
+        match result_2 {
+            Ok(()) => {}
+            Err(e) => panic!("Error making receipt request: {:?}", e),
+        }
+    }
+    Ok(())
+}
+
+#[rstest]
+#[tokio::test]
+async fn test_manager_wrong_aggregator_keys(
+    #[future] single_indexer_wrong_gateway_test_server: Result<
+        (ServerHandle, SocketAddr, ServerHandle, SocketAddr),
+        Error,
+    >,
+    #[future] requests_1: Result<Vec<(EIP712SignedMessage<Receipt>, u64)>>,
+    receipt_threshold_1: u64,
+) -> Result<()> {
+    let (_server_handle, socket_addr, _gateway_handle, _gateway_addr) =
+        single_indexer_wrong_gateway_test_server.await?;
+    let indexer_1_address = "http://".to_string() + &socket_addr.to_string();
+    let client_1 = HttpClientBuilder::default().build(indexer_1_address)?;
+    let requests = requests_1.await?;
+
+    let mut counter = 1;
+    for (receipt_1, id) in requests {
+        let result = client_1.request("request", (id, receipt_1)).await;
+        if (counter % receipt_threshold_1) == 0 {
+            match result {
+                Ok(()) => panic!("Should have failed signature verification"),
+                Err(_) => {}
+            }
+        } else {
+            match result {
+                Ok(()) => {}
+                Err(e) => panic!("Error making receipt request: {:?}", e),
+            }
+        }
+        counter += 1;
+    }
+
+    Ok(())
+}
+
+#[rstest]
+#[tokio::test]
+async fn test_manager_wrong_requestor_keys(
+    #[future] single_indexer_test_server: Result<
+        (ServerHandle, SocketAddr, ServerHandle, SocketAddr),
+        Error,
+    >,
+    #[future] wrong_requests: Result<Vec<(EIP712SignedMessage<Receipt>, u64)>>,
+    receipt_threshold_1: u64,
+) -> Result<()> {
+    let (_server_handle, socket_addr, _gateway_handle, _gateway_addr) =
+        single_indexer_test_server.await?;
+    let indexer_1_address = "http://".to_string() + &socket_addr.to_string();
+    let client_1 = HttpClientBuilder::default().build(indexer_1_address)?;
+    let requests = wrong_requests.await?;
+
+    let mut counter = 1;
+    for (receipt_1, id) in requests {
+        let result = client_1.request("request", (id, receipt_1)).await;
+        if (counter % receipt_threshold_1) == 0 {
+            match result {
+                Ok(()) => panic!("Should have failed signature verification"),
+                Err(_) => {}
+            }
+        } else {
+            match result {
+                Ok(()) => {}
+                Err(e) => panic!("Error making receipt request: {:?}", e),
+            }
+        }
```

Comment:
Please add comments explaining the if/else blocks

Reply:
added comments